src/OrleansRuntime/Silo/Silo.cs

```diff
@@ -178,6 +178,7 @@ internal Silo(SiloInitializationParameters initializationParams, IServiceProvide
                 var serviceCollection = new ServiceCollection();
                 serviceCollection.AddSingleton<Silo>(this);
                 serviceCollection.AddSingleton(initializationParams);
+                serviceCollection.AddLegacyClusterConfigurationSupport(config);
```

Comment:
Isn't this should be in SiloHostBuilder.UseConfiguration method ? Why is it here?

Reply:
Because this is the path where the user doesn't use `SiloBuilder`, and instead instantiates `Silo` directly (or via `AzureSilo`, or `SiloHost`). So this path effectively ONLY supports legacy configuration. This path will probably go away very soon, @ReubenBond was bringing it up as something to get rid of for 2.0, and I think I agree, but of course need to be dealt separately as just dropping support in this PR should not be by itself a breaking change. We are tracking it for discussion for post 2.0-beta

Reply:
ah nvm . you added it there